trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisSystemRepository.java

```diff
@@ -75,4 +75,14 @@ SELECT id, name, elite_id, ST_X(coordinates_geom) as x_coordinate, ST_Y(coordina
     })
     @ResultMap("systemResultMap")
     MybatisSystemEntity createOrUpdateOnConflict(MybatisSystemEntity map);
+    
+    @Select({
+            "SELECT id, name, elite_id, ST_X(coordinates_geom) as x_coordinate, ST_Y(coordinates_geom) as y_coordinate, ST_Z(coordinates_geom) as z_coordinate " +
+                    "FROM system " +
+                    "WHERE name ILIKE CONCAT('%', #{name}, '%')" +
+                    "ORDER BY CASE WHEN name ILIKE CONCAT(#{name}, '%') THEN 0 ELSE 1 END, name " +
+                    "LIMIT #{amount}"
+    })
+    @ResultMap("systemResultMap")
+    List<MybatisSystemEntity> findSystemsByNameContaining(@Param("name") String name, @Param("amount") Integer amount);
```

Comment:
Could you performance test this query as is vs w/o a leading wildcard? I don't think it's appropriate to search for systems by name containing. Instead would it not make more sense to search by name beginning with? Open to discussion here. For reference, using Insomnia to query against `https://duval.edpn.io/api/v1/exploration/system/by-name-containing?subString=Sol&amount=10` takes me from 600ms to 1s.

Reply:
go ahead and hit that endpoint again, ill push a separate command to add it, but I just threw a temporary index on it that brings query time down to <10ms so its all api time to present now. Let me know if this is fast enough, I can curl it in half a second.

Reply:
now getting in the 130ms range. much better. thanks